src/app/payment/views/IntegratedCheckoutView/CheckoutViewWrapper.tsx

```diff
@@ -432,39 +448,18 @@ const CheckoutViewWrapper = () => {
     setAuthMethod('signUp');
   };
 
-  const getUserInfo = ({
-    email,
-    userNameFromAddressElement,
-    fullName,
-    user,
-  }: {
-    email: string;
-    userNameFromAddressElement: string;
-    fullName: string;
-    user?: UserSettings;
-  }) => {
-    let userData;
-
-    if (user) {
-      userData = {
-        name: fullName,
-        email: user.email,
-      };
-    } else {
-      userData = {
-        name: userNameFromAddressElement,
-        email: email,
-      };
-    }
-
-    return userData;
-  };
-
   const handlePromoCodeError = (err: unknown, showNotification?: boolean) => {
     const error = err as Error;
-    const errorMessage = error.message.includes('Promotion code with an id')
-      ? error.message
-      : 'Something went wrong, try again later';
+    const statusCode = (err as any).status;
+    let errorMessage = 'Something went wrong, try again later';
```

Comment:
A more specific message would be better, so that it is known that the promo code is not valid, and it should be added to the translations